DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/svd/CMakeLists.txt

```
@@ -64,38 +43,17 @@ else()
     endif()
 endif()
 
-if(NOT DEFINED DEVICE_FLAG)
-    message(FATAL_ERROR "An unrecognized or custom board was passed, but DEVICE_FLAG was not specified. \
-                         Make sure you have set -DDEVICE_FLAG=A10, -DDEVICE_FLAG=S10 or -DDEVICE_FLAG=Agilex7")
-endif()
-
-set(BENCHMARK 1)
-if(DEVICE_FLAG MATCHES "A10")
-    # A10 parameters
-    set(FIXED_ITERATIONS 50)
-elseif(DEVICE_FLAG MATCHES "S10")
-    # S10 parameters
-    set(FIXED_ITERATIONS 55)
-elseif(DEVICE_FLAG MATCHES "Agilex7")
-    # Agilex 7 parameters
+if(NOT DEFINED FIXED_ITERATIONS)
     set(FIXED_ITERATIONS 55)
-else()
-    message(FATAL_ERROR "Unreachable")
-endif()
-
-if(DEFINED SET_BENCHMARK)
-    set(BENCHMARK ${SET_BENCHMARK})
-endif()
-
-if(DEFINED SET_FIXED_ITERATIONS)
-    set(FIXED_ITERATIONS ${SET_FIXED_ITERATIONS})
 endif()
-
-message(STATUS "BENCHMARK=${BENCHMARK}")
 message(STATUS "FIXED_ITERATIONS=${FIXED_ITERATIONS}")
 
+# Use cmake -DUSER_FPGA_FLAGS=<flags> to set extra flags for FPGA backend
+# compilation. 
+set(USER_FPGA_FLAGS ${USER_FPGA_FLAGS})
+
 # Use cmake -DUSER_FLAGS=<flags> to set extra flags for general compilation.
-set(USER_FLAGS ${USER_FLAGS})
+set(USER_FLAGS ${USER_FLAGS} -DFIXED_ITERATIONS=${FIXED_ITERATIONS} ${BSP_FLAG})
```

```suggestion
set(USER_FLAGS ${USER_FLAGS};-DFIXED_ITERATIONS=${FIXED_ITERATIONS};${BSP_FLAG})
```
cmake flags like this should be separated by semicolons